diploma-project/components/ProductCard.jsx

```diff
@@ -1,33 +1,91 @@
 import Image from "next/future/image";
+import {default_profile_picture} from "../public/constants";
+import {useEffect, useState} from "react";
+import {getUserById, User} from "../services/user";
+import {Timestamp} from "firebase/firestore";
+import {useRouter} from "next/router";
+import Link from "next/link";
+
+const ProductCard = ({product, inCatalog = false, productId}) => {
+    const [isNew, setIsNew] = useState(false);
+    const [owner, setOwner] = useState(new User());
+    const [isLiked, setIsLiked] = useState(false);
+    const router = useRouter();
+    const tags = ["silver craft", "wood craftttttttttt"];
+
+    useEffect(() => {
```

Comment:
The getUserById function is called in the useEffect hook without any dependencies. This means it will only be called once when the component mounts. If the product.owner prop changes, the user won't be fetched again. Consider if you want to add product.owner to the dependencies array of the useEffect hook, so that it can update.

Reply:
I don't think an owner of a product should change. That is why I haven't added any dependencies to the useEffect hook.